### PR TITLE
framebuf, text + test

### PR DESCRIPTION
**Pull request by ladyada**
allow use of adafruit_framebuf if framebuf isnt't available
tested text in the adafruit_framebuf implementation
add a more fully-featured test for framebuf usage

### CODE REVIEW

**Comment by eighthree** [comment]
Hello! I've tested this on a Feather M4 Express with no issues. Example and library seem to work well.

On a trinket m0 I get this error running ```ssd1306_framebuftest.py``` as code.py:
```
Auto-reload is on. Simply save files over USB to run them or enter REPL to disable.
code.py output:
Framebuf capability test - these are slow and minimal but don't requirea special graphics management library, only `adafruit_framebuf`
Pixel test
Lines test
Traceback (most recent call last):
  File "code.py", line 55, in <module>
TypeError: function takes 1 positional arguments but 6 were given
```

Line 55 looks like:
```display.line(corner_from[0], corner_from[1],  corner_to[0], corner_to[1], 1)```

Earlier today I modified a version of the adafruit_ssd1306 library to test the new adafruit_framebuf library, with only a few line changes:
https://gist.github.com/eighthree/232110a6a35e0d8677eddaea2055560e
This version makes ```ssd1306_framebuftest.py``` work without any modifications.

I reckon the error is something else other than what the error message indicates.

**Comment by jerryneedell** [comment]
The error reported above occurred for me when I had the old "framebuf.py" module on my system. After removing it, it worked.  On trinkets, the old framebuf was necessary since the builtin framebuf was not included,

**Comment by jerryneedell** [comment]
I have tested this on a metro_m4_espress wit ha 128x32 featherwing.  For the ssd1306_framebuftest.py I had to change the address to 3c and set the resetpin to None.

**Comment by jerryneedell** [comment]
worked on my trinket_m0 with oled featherwing.

File: adafruit_ssd1306.py [hunk 1]
```diff
@@ -63,7 +66,7 @@ class _SSD1306:
     """Base class for SSD1306 display driver"""
     #pylint: disable-msg=too-many-arguments
     #pylint: disable-msg=too-many-instance-attributes
-    def __init__(self, framebuffer, width, height, external_vcc, reset):
+    def __init__(self, framebuffer, width, height, *, external_vcc, reset):
```
Comment (jerryneedell) [inline adafruit_ssd1306.py]: since reset is an optional keyword does it default to None? if not should it be explicitly set to None as the default?

**Review by jerryneedell — COMMENTED** [review]
Tested on a metro_m4_espress  and trinket_m0 with an 128x32 oled featherewing -- both ran the test cases without error.  Also ran some "old" ssd1306 code on the metro_m4 -- no issues.
Looks good to me.

**Review by jerryneedell — APPROVED** [review]
also tested SPI 132x28 OLED with metro_m4_express -- once I got the pins defined right it was fine ;-)

**Comment by jerryneedell** [comment]
for some reason  - my review went in on behalf of circuitpython librarians. I added @adafruit/circuitpythonlibrarians back as a reviewer

**Comment by ladyada** [comment]
thanx, yah this superscedes the old framebuf, and i didnt keep the same name since there might be some structural changes

**Pull request merged**